dronegen/container_images.go

```diff
@@ -0,0 +1,942 @@
+// 5. Set `testingQuayRegistryOrg` and `testingECRRegistryOrg` to the org name(s) used in (4)
+// 6. Set the `ECRTestingDomain` to the domain used for the private ECR repos
+// 7. Create two separate IAM users, each with full access to either the public ECR repo OR the private ECR repo
+// 8. Create a Quay "robot account" with write permissions for the created Quay repos
+// 9. Set the Drone secrets for the secret names listed in "GetContainerRepos" to the credentials in (7, 8), prefixed by the value of `testingSecretPrefix`
+//
+// On each commit, after running `make dronegen``, run the following commands and resign the file:
+// # Pull the current branch instead of v10 so the appropriate dockerfile gets loaded
+// sed -i '' "s~git checkout -qf \"\$(cat '/go/vars/full-version/v10')\"~git checkout -qf \"${DRONE_SOURCE_BRANCH}\"~" .drone.yml
+//
+// When finishing up your PR check the following:
+// * The testing secrets added to Drone have been removed
+// * `configureForPRTestingOnly` has been set to false, and `make dronegen` has been reran afterwords
+
+const (
+	configureForPRTestingOnly bool   = false
+	testingSecretPrefix       string = "TEST_"
+	testingQuayRegistryOrg    string = "fred_heinecke"
+	testingECRRegistryOrg     string = "u8j2q1d9"
+	testingECRRegion          string = "us-east-2"
+	prBranch                  string = "fred/multiarch-teleport-container-images"
+	testingECRDomain          string = "278576220453.dkr.ecr.us-east-2.amazonaws.com"
+)
+
+const (
+	ProductionRegistryOrg string = "gravitational"
+	PublicEcrRegion       string = "us-east-1"
+	StagingEcrRegion      string = "us-west-2"
+
+	localRegistry string = "drone-docker-registry:5000"
+)
+
+func buildContainerImagePipelines() []pipeline {
+	// *************************************************************
+	// ****** These need to be updated on each major release. ******
+	// *************************************************************
+	latestMajorVersions := []string{"v10", "v9", "v8"}
+	branchMajorVersion := "v10"
+
+	triggers := []*TriggerInfo{
+		NewTagTrigger(branchMajorVersion),
+		NewPromoteTrigger(branchMajorVersion),
+		NewCronTrigger(latestMajorVersions),
+	}
+
+	if configureForPRTestingOnly {
+		triggers = append(triggers, NewTestTrigger(prBranch, branchMajorVersion))
+	}
+
+	pipelines := make([]pipeline, 0, len(triggers))
+	for _, trigger := range triggers {
+		pipelines = append(pipelines, trigger.buildPipelines()...)
+	}
+
+	return pipelines
+}
+
+type TriggerInfo struct {
+	Trigger           trigger
+	Name              string
+	Flags             *TriggerFlags
+	SupportedVersions []*releaseVersion
+	SetupSteps        []step
+}
+
+// This is mainly used to make passing these vars around cleaner
+type TriggerFlags struct {
+	ShouldAffectProductionImages bool
+	ShouldBuildNewImages         bool
+	UseUniqueStagingTag          bool
+}
+
+func NewTestTrigger(triggerBranch, testMajorVersion string) *TriggerInfo {
```

Comment:
Is this just for testing and need to be removed before merge?

Reply:
It is just for testing but I would recommend it stays after merge to allow for easier testing in the future. Basically this (and the block of testing const you commented on below) make it significantly easier to test changes without affecting production repos. This issue stems from not having a "staging" or "testing" drone instance to test PRs on. 

Would you prefer I remove these sections?